Add shift key details to kb shortcuts documentation (#2725)
OfficeDev/office-js-docs-pr: docs/design/keyboard-shortcuts.md

- The property names `action`, `key`, and `default` are required.
- The value of the `action` property is a string and must match one of the `id` properties in the action object.
- The `default` property can be any combination of the characters A - Z, a -z, 0 - 9, and the punctuation marks "-", "_", and "+". (By convention, lower case letters are not used in these properties.)
- The `default` property must contain the name of at least one modifier key (Alt, Ctrl, Shift) and only one other key. 

> - Shift cannot be used as the only modifier key. If using Shift, you must also use at least one other modifier key (Alt, Ctrl).

Reviewer: Suggestion for brevity:
```suggestion
- Shift cannot be used as the only modifier key. Combine Shift with either Alt or Ctrl.
```
